### ISSUE TITLE
Cannot import environment file at startup

### ISSUE THREAD

**Issue by fantoine** [Unleash/unleash#8404]
Hi there,

I try to implement the auto-import of feature flags on a local dev environment using the `IMPORT_FILE` env configuration but I keep getting the following issue at Unleash startup:

```
 [ERROR] TypeError: Cannot read properties of undefined (reading 'filter')
     at ExportImportService.removeArchivedFeatures (/unleash/node_modules/unleash-server/dist/lib/features/export-import-toggles/export-import-service.js:309:63)
     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
     at async ExportImportService.cleanData (/unleash/node_modules/unleash-server/dist/lib/features/export-import-toggles/export-import-service.js:279:36)
     at async ExportImportService.importFromFile (/unleash/node_modules/unleash-server/dist/lib/features/export-import-toggles/export-import-service.js:121:28)
     at async createApp (/unleash/node_modules/unleash-server/dist/lib/server-impl.js:111:9)
     at async Object.start (/unleash/node_modules/unleash-server/dist/lib/server-impl.js:165:21)
```

I use a docker/docker-compose setup (as described [here](https://docs.getunleash.io/how-to/how-to-environment-import-export#startup-import)) and tried to follow the schema described on the [API doc](https://docs.getunleash.io/reference/api/unleash/import-toggles) to prepare the JSON file to import.

For test purposes, I reduced it to a very small sample:

```json
{
  "features": [
    {
      "name": "kill-switch-my-feature",
      "description": "Some feature...",
      "type": "kill-switch",
      "enabled": true,
      "environments": [{
        "name": "development",
        "enabled": true,
        "strategies": [{
          "name": "default"
        }]
      }]
    }
  ]
}
```

The docker-compose configuration looks like this for the unleash part:

```yaml
#...
  unleash:
    image: unleashorg/unleash-server:6.3.0
    restart: always
    ports:
      - '4242:4242'
    volumes:
      - './feature-flags.json:/feature-flags.json'
    environment:
      DATABASE_HOST: 'postgres'
      DATABASE_PORT: 5432
      DATABASE_USERNAME: 'unleash'
      DATABASE_PASSWORD: 'unleash'
      DATABASE_NAME: 'unleash'
      DATABASE_SSL: 'false'
      IMPORT_FILE: '/feature-flags.json'
      IMPORT_PROJECT: 'default'
      IMPORT_ENVIRONMENT: 'development'
    depends_on:
      - postgres
#...
```

After some investigation, it looks like the error comes from this file: https://github.com/Unleash/unleash/blob/main/src/lib/features/export-import-toggles/export-import-service.ts#L696 where `dto.data.featureStrategies` nullability is not checked while it can be skipped (according to schema described on the API).

<img width="675" alt="image" src="https://github.com/user-attachments/assets/0d9c1f27-0034-42b7-9901-5a47f62b4b05">

As a quick workaround, I tried to add a `featureStrategies` field on my json with an empty array value, but I get an other error since the system user used to import files at startup doesn't have rights to manage strategies.

I wonder if the solution is to fix the nullability check on this field, or if there is any other way to make it working ?

**Comment by sjaanus** [comment]
@fantoine, thank you for reporting this.

You mentioned that, based on the API schema, `featureStrategies` can be optional. However, in the schema, `dto.data.featureStrategies` appears to be required.

Providing an empty array for this field should resolve the issue.

![Image](https://github.com/user-attachments/assets/ce48446a-b92f-43af-bc61-544fcd3b54b7)

**Comment by sjaanus** [comment]
To be honest I am a little puzzled how did you even get that far, because when I try to import I get this immediately from schema validator.

```
{
    "id": "76641ffe-ab00-4ee5-983a-2843ab03a735",
    "name": "BadDataError",
    "message": "Request validation failed: your request body or params contain invalid data. Refer to the `details` list for more information.",
    "details": [
        {
            "path": "/body/data/featureStrategies",
            "message": "The `/body/data/featureStrategies` property is required. It was not present on the data you sent."
        }
    ]
}

**Comment by fantoine** [comment]
Hi @sjaanus !

Thanks for your quick answer.

To be honest, I'm not really sure about what happened on this case...

Before creating this issue, I also tried to add an empty `featureStrategies` array but I got an other error mentioning that  unleash_system_user didn't have rights to manage strategies...

Today, I retried with a clean setup (Docker image and unleash database) and using this empty array, and now it works fine... :/

Well... it may comes from some wrong state on my environment...

I guess that the issue can be closed. Thanks for your time.
